src/logging.ts

```diff
@@ -0,0 +1,40 @@
+import debug, { Debugger } from 'debug';
+
+const globalLogger = debug('metamask');
+
+/**
+ * Creates a logger via the `debug` library whose log messages will be tagged
+ * using the name of your project. By default, such messages will be
+ * suppressed, but you can reveal them by setting the `DEBUG` environment
+ * variable to `metamask:<projectName>`. You can also set this variable to
+ * `metamask:*` if you want to see log messages from all MetaMask projects that
+ * are also using this function to create their loggers.
+ *
+ * @param projectName - The name of your project. This should be the name of
+ * your NPM package if you're developing one.
+ * @returns An instance of `debug`.
+ */
+export function createProjectLogger(projectName: string): Debugger {
+  return globalLogger.extend(projectName);
```

Comment:
I'm choosing to wrap `debug` in case we want to swap this out for something else in the future.